gossipsub.go

```diff
@@ -1732,7 +1749,7 @@ func (gs *GossipSubRouter) piggybackControl(p peer.ID, out *RPC, ctl *pb.Control
 }
 
 func (gs *GossipSubRouter) makePrune(p peer.ID, topic string, doPX bool) *pb.ControlPrune {
-	if gs.peers[p] == GossipSubID_v10 {
+	if gs.peerHasAny(p, GossipSubID_v10) {
```

Comment:
I think the right approach here is to check for the PX feature, and passing the protocol.
So `gs.SupportsFeature(GossipSubFeaturePX, p)` and so on.